nibabel/streamlines/utils.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+import nibabel
+
+
+def get_affine_from_reference(ref):
+    """ Returns the affine defining the reference space.
+
+    Parameter
+    ---------
+    ref : str or :class:`Nifti1Image` object or ndarray shape (4, 4)
+        If str then it's the filename of reference file that will be loaded
+        using :func:nibabel.load in order to obtain the affine.
+        If :class:`Nifti1Image` object then the affine is obtained from it.
+        If ndarray shape (4, 4) then it's the affine.
+
+    Returns
+    -------
+    affine : ndarray (4, 4)
+        Transformation matrix mapping voxel space to RAS+mm space.
+    """
+    if type(ref) is np.ndarray:
```

Comment:
Can you avoid the type check here?  Maybe with `if hasattr(ref, 'shape')` or similar?   Sometimes people duck-type arrays...